packages/orbit-components/src/Toast/index.jsx.flow

```diff
@@ -0,0 +1,55 @@
+// @flow
+/*
+  DOCUMENTATION: https://orbit.kiwi/components/toast/
+*/
+import * as React from "react";
+
+import type { Globals } from "../common/common.js.flow";
+
+export type Placement =
+  | "top-left"
+  | "top-center"
+  | "top-right"
+  | "bottom-left"
+  | "bottom-center"
+  | "bottom-right";
+
+export type Toast = {|
+  +icon?: React.Element<any>,
+  +visible?: boolean,
+  +children: React.Node,
+  +dismissTimeout?: number,
+  +onMouseEnter: () => void,
+  +onMouseLeave: () => void,
+  +onDismiss: () => void,
+  +placement: Placement,
+  +offset?: number,
+  +ariaLive?: string,
+  +role?: string,
+|};
+
+export type Props = {|
+  +topOffset?: number,
+  +leftOffset?: number,
+  +rightOffset?: number,
+  +bottomOffset?: number,
+  +gutter?: number,
+  +dismissTimeout?: number,
+  +placement?: Placement,
+  ...Globals,
+|};
+
+export type createToastType = (message: string, options: {| icon?: React.Element<any> |}) => void;
+export type createToastPromiseType = (
+  promise: Promise<void>,
+  options: {|
+    icon?: React.Element<any>,
+    loading?: React.Node,
+    success?: string | ((data: any) => string),
+    error?: string | ((err: any) => string),
+  |},
+) => void;
+
+declare export default React.ComponentType<Props>;
+declare export var createToast: createToastType;
+declare export var createToastPromise: createToastPromiseType;
```

Comment:
Considering the amount of types and the importance of Toast it would be better to write `__typetests__` for TypeScript to ensure that there isn't some major mistake.

Reply:
why? `Toast` is not supposed to be used directly by users. Users will use only `ToastInit` and `createToast` which accepts very few arguments.

Reply:
Never mind now.